### ISSUE TITLE
Feature Suggestions

### ISSUE THREAD

**Issue by ldsands** [dotnet/corefxlab#2864]
I'm excited about this work! I've been using python for a while and while overall a vast improvement over R for my data analysis, it can still be lacking.

FYI I'm an academic with almost no formal training in computer science. Please forgive me for any opinions I have that are uninformed. I think my feature requests would get you a lot more interest from the academic community.

### Increased IO support?

I think that if you want a lot of people to try it out the more formats, you support the more people will try it out. File IO is an easy excuse to not try out alternatives especially when there are larger files.

These file formats would be very handy for getting attention from a lot academics if you had them all (in order of my preference)?

1. .parquet ideally (you'll need gzip, snappy support and I'm personally a big fan of zstd compression for parquet). [Here is a repo that deals with parquet files](https://github.com/aloneguid/parquet-dotnet)
1. json
1. rdata, .rda for R
1. [Stata .dta](https://www.stata.com/help.cgi?dta) format a lot social scientists use Stata. I think Stata is ripe for a quick death if someone else can make a lot of effort to make doing statistics easy like stata is.
1. feather (my understanding is that is a lot like parquet but quick and not meant for long-term storage)

### Statistics built in or as a side package that is easy to access

Or at least included in some future documentation. For example you could make it very easy to do basic correlations, ANOVAS, linear and logistic regression eceterea. I would like to also see the opportunity to pull out pretty much any part of the calcuations along with a lot of accompaning staistics. Also, being able to output the resutls in an [easy to read format would help a lot too](https://feliperego.github.io/blog/2015/10/23/Interpreting-Model-Output-In-R).

Again this is all from a very academic perspective if that is not who you're targeting I completely understand. Either way good luck!

**Comment by ldsands** [comment]
I almost forgot. Missing values I saw some discussion about if you people wanted missing values to be the same across different data types. I think it would be fine if there were options for both that would be fine. However, again as a non-computer scientist having several kinds of missing values for different columns is a major pain! [Pandas has realized this too and are working towards a unified(ish) missing type](https://jorisvandenbossche.github.io/blog/2019/11/30/pandas-consistent-missing-values/). I would really like to see something similar here as well. If I don't what to see the type difference for the missing values that would save me a ton of time. My data is never as clean as a lot of businesses deal with, missing values are everywhere so I just have to deal with them. This makes my life a lot easier. R and Julia both do this as well.

Again, good luck!

**Comment by pgovind** [comment]
Great feedback! About the IO file support, I'm thinking about starting a DataFrameIO project that deals with the different IO formats. My hope is that we can seed a few formats and the community can contribute to the other ones.

Can you elaborate more on the different types of missing values you are encountering? AFAIK, we support missing values for all column types the same way in DataFrame: as `null` values. So in our case, both a float column and an int column will have `null` values representing missing values. Any operations on them will have effect on the column type. This does mean that we don't differentiate between `null` values and `missing` values though. Is that part of what you mean?

**Comment by ldsands** [comment]
First off, I think the separate DataFrameIO project is a great idea. It never ceases to amaze me how difficult IO can be when trying out some languages. The more this gets streamlined and easy to deal with, the faster the cleaning and analysis can happen.

As for the missing values, I wasn't really sure I tried looking through your documentation and I must have misinterpreted something. It looks like you're already doing what I was talking about. Though just to make sure, a null would work for a string or bool or all other types of columns too, right?

**Comment by pgovind** [comment]
Yup. `null` works for string, bool and the numeric column types. There's actually strong unit tests for this :)

**Comment by ldsands** [comment]
Perfect! And thank you for your patience with my lack of experience with c#/dotnet etc.

**Comment by plaisted** [comment]
With DataFrames being based on Arrow, first class support for Parquet would be great. This is currently accomplished in dotnet/spark with Spark (java/scala) doing the Parquet -> Arrow conversion. Being able to use dataframes + parquet locally (ie run non-distributed jobs) using the same implementation (on dataframes) would be open up a lot of use cases.
